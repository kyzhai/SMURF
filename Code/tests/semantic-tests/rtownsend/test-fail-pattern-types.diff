1c1
< Fatal error: exception Sast.Type_mismatch("ftrue")
---
> Fatal error: exception Sast.Type_mismatch("Patterns don't match type signature for f true")
