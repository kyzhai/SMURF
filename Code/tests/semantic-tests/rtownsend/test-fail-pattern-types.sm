/* The types of the patterns for a function declaration must match with their corresponding
	 types in the type signature for that function */

f :: Int -> Int

f True = 0

main = []
