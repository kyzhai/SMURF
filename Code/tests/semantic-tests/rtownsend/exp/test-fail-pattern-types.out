Fatal error: exception Sast.Type_mismatch("Patterns don't match type signature for f true")
